### PR TITLE
AE->OSO ansible uninstall merge

### PR DESCRIPTION
**Pull request by aveshagarwal**
playbooks/adhoc: Add a tutorial-reset playbook to undo everything

This makes it easier to run through the tutorial, as well as reset
a VM or baremetal node to a clean slate for developer testing.

### CODE REVIEW

**Comment by aveshagarwal** [comment]
@detiber @sdodson

File: playbooks/adhoc/tutorial-reset.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      changed_when: False
+
+    - file: path={{ item }} state=absent
+      with_items:
+        - /var/lib/atomic-enterprise
+        - /etc/sysconfig/atomic-enterprise
+        - /etc/atomic-enterprise
+        - /etc/openshift
+        - /var/lib/docker
+
```
Comment (sdodson) [inline playbooks/adhoc/tutorial-reset.yml]: How do you feel about removing `/root/.kube`? cruft there seems to trip a lot of people up and may prevent re-installation.

Comment (aveshagarwal) [inline playbooks/adhoc/tutorial-reset.yml]: @sdodson +1 will update the PR.

**Comment by aveshagarwal** [comment]
Pushed updated PR.

**Comment by sdodson** [comment]
Thanks, LGTM.

**Comment by detiber** [comment]
@aveshagarwal only critique I have is that the adhoc playbook name should probably be atomic_enterprise_tutorial_reset.yml ('_' are used more commonly in ansible file naming than '-' most likely due to '_' being a valid character in a python variable, and including atomic enterprise in the name, since it doesn't apply to the openshift tutorial).

**Comment by aveshagarwal** [comment]
@detiber I agree with you. will update the PR.

**Comment by ashcrow** [comment]
@detiber / @sdodson mind taking a look to see if this can be merged today?

**Comment by detiber** [comment]
:+1: @brenton ready for merge.

**Comment by brenton** [comment]
Does anyone know what the experience will be for admins if ansible removes /var/lib/docker and they are using `docker-storage-setup`?  I think for lvm thinpools will lead to an error on docker startup saying something about being unable to claim ownership of the pool.  I have no idea what it would do if normal volume groups are being used.

**Comment by aveshagarwal** [comment]
@brenton I understand your concern, but it was created mainly to cleanup and start afresh when someone is going through the tutorial and makes mistakes and gets stuck.

**Comment by ashcrow** [comment]
@brenton / @aveshagarwal is this a test case that needs to be checked before merge?

**Comment by detiber** [comment]
@ashcrow @brenton @aveshagarwal What if instead of rm -rf /var/lib/docker, it checked for lvm-direct instead of lvm-loop (can be derived from docker info), if lvm-loop, kill /var/lib/docker...  if lvm-direct just docker rm all contianers and docker rmi all images.

Thoughts?

**Comment by aveshagarwal** [comment]
@detiber sounds good to me.

**Comment by brenton** [comment]
@ashcrow, we don't have a test case for this today.  However, there is definitely a card for this work to eventually be done so we can have QE try this out once it's ready!

**Comment by aveshagarwal** [comment]
@brenton @ashcrow @detiber  @sdodson  this was only created for resetting AE tutorial. So if the plan is now to increase its scope to cover all products, my suggestion would be that someone from productization or QE take its ownership.

**Comment by ashcrow** [comment]
@brenton makes sense. It sounds like that would be an enhancement beyond the scope of @aveshagarwal's work.

**Comment by aveshagarwal** [comment]
@brenton @detiber @sdodson @ashcrow  updated PR.

 It updates tutorial reset file and makes following chages:

1. Included openshift clean up
2. Renamed file to atomic_openshift_tutorial_reset.yml
3. docker service is not not stopped
4. docker containers and images are removed
5. /etc/openshift-sdn are removed too now
6. /var/ib/docker is not removed

**Comment by aveshagarwal** [comment]
just FYI, I also tested it with the latest openshift-ansible and openshift and it worked.

**Comment by aveshagarwal** [comment]
Also now both products openshift and atomic-enterprise can be reset with this.

File: playbooks/adhoc/atomic_openshift_tutorial_reset.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        - atomic-enterprise
+        - atomic-enterprise-master
+        - atomic-enterprise-node
+        - atomic-enterprise-sdn-ovs
+        - tuned-profiles-atomic-enterprise-node
+        - openshift
+        - openshift-master
+        - openshift-node
+        - openshift-sdn-ovs
+        - tuned-profiles-openshift-node
```
Comment (sdodson) [inline playbooks/adhoc/atomic_openshift_tutorial_reset.yml]: May as well add `origin` `origin-master` `origin-node` `origin-sdn-ovs` and `tuned-profiles-origin-node`

File: playbooks/adhoc/atomic_openshift_tutorial_reset.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    - file: path={{ item }} state=absent
+      with_items:
+        - /var/lib/atomic-enterprise
+        - /etc/sysconfig/atomic-enterprise
+        - /etc/atomic-enterprise
+        - /etc/openshift
+        - /etc/openshift-sdn
+        - /root/.kube
+        - /etc/sysconfig/openshift
+        - /var/lib/openshift
```
Comment (sdodson) [inline playbooks/adhoc/atomic_openshift_tutorial_reset.yml]: /var/lib/origin
/etc/origin

Comment (sdodson) [inline playbooks/adhoc/atomic_openshift_tutorial_reset.yml]: And I guess the sysconfig files should be
/etc/sysconfig/origin-master
/etc/sysconfig/origin-node
/etc/sysconfig/openshift-master
/etc/sysconfig/openshift-node
/etc/sysconfig/atomic-openshift-master
/etc/sysconfig/atomic-openshift-node

Comment (aveshagarwal) [inline playbooks/adhoc/atomic_openshift_tutorial_reset.yml]: @sdodson thanks will make those chages. Could you please services files name too if they are in line with the recent naming. If not I will update everything at once.

Comment (aveshagarwal) [inline playbooks/adhoc/atomic_openshift_tutorial_reset.yml]: @sdodson also what about /etc/openshift-sdn, would it stay same for all products?

File: playbooks/adhoc/atomic_openshift_tutorial_reset.yml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  sudo: yes
+
+  tasks:
+    - service: name={{ item }} state=stopped
+      with_items:
+        - openshift-master
+        - openshift-node
+        - openvswitch
+        - atomic-enterprise-master
+        - atomic-enterprise-node
```
Comment (sdodson) [inline playbooks/adhoc/atomic_openshift_tutorial_reset.yml]: atomic-openshift-master
atomic-openshift-node

File: playbooks/adhoc/atomic_openshift_tutorial_reset.yml (hunk 3 above)
Comment (sdodson) [inline playbooks/adhoc/atomic_openshift_tutorial_reset.yml]: @aveshagarwal Yeah that remains the same across all products.

File: playbooks/adhoc/atomic_openshift_tutorial_reset.yml (hunk 4 above)
Comment (aveshagarwal) [inline playbooks/adhoc/atomic_openshift_tutorial_reset.yml]: @sdodson  Based on https://github.com/sdodson/origin/blob/package-refactor, I have come up with following:

1. origin, AE and openshift
 /etc/origin same for all 
what about /var/lib/origin? would it stay same for  openshift and AE or would it be /var/lib/atomic-openshift?

2. origin rpms: 
origin, origin-master, origin-node, origin-sdn-ovs, tuned-profiles-origin-node 

AE and openshift rpms
        - atomic-openshift
        - atomic-openshift-master
        - atomic-openshift-node
        - atomic-openshift-sdn-ovs
        - tuned-profiles-atomic-openshift-node

3. origin service files:
origin-master, origin-node

AE and openshift services:
        - atomic-openshift-master
        - atomic-openshift-node

4. origin /etc/sysconfig
        - /etc/sysconfig/origin-master
        - /etc/sysconfig/origin-node

AE and openshift:
        - /etc/sysconfig/atomic-openshift-master
        - /etc/sysconfig/atomic-openshift-node

5. binaries:
origin: /bin/origin
AE: /bin/atomic-enterprise
openshift: /bin/openshift

Comment (ashcrow) [inline playbooks/adhoc/atomic_openshift_tutorial_reset.yml]: The statedir for both is named origin: ```%{_sharedstatedir}/origin``` which is ```/var/lib/origin```

Ref: https://github.com/openshift/origin/pull/4128/files#diff-3477fbcab168ba3ee773ffeb9323e53eR243

**Comment by aveshagarwal** [comment]
@sdodson @detiber @brenton pushed updated PR

Updated to include origin and atomic-openshift RPMs re-factoring to include all origin, AE and openshift products. For back-word compatibility, older openshift and AE naming is retained too.

**Comment by sdodson** [comment]
:+1: I think we should merge as is, I'll file a followup issue to make this a more generic uninstall procedure that covers more comprehensive set of uninstall tasks

**Comment by aveshagarwal** [comment]
@ashcrow I already pointed out that for back word compatibility or older installs.

**Comment by ashcrow** [comment]
@aveshagarwal understood. If they are there only for backwards compatibility I see no problem.

**Comment by aveshagarwal** [comment]
@sdodson with the issue https://github.com/openshift/openshift-ansible/issues/485, would you want to merge this PR now or wait for the issue to be resolved/addressed before going forward with the merge. I am fine with either.

**Comment by sdodson** [comment]
I'd say merge this and we'll address #485 later

**Comment by aveshagarwal** [comment]
ok sounds good.

**Pull request merged**